src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs

```diff
@@ -42,5 +42,14 @@ public int ScrapeResponseCacheDurationMilliseconds
         set => this.ExporterOptions.ScrapeResponseCacheDurationMilliseconds = value;
     }
 
+    /// <summary>
+    /// Gets or sets a value indicating whether to export OpenMetrics compatible scrape responses. Default value: true.
+    /// </summary>
+    public bool OpenMetricsEnabled
```

Comment:
Consider a different approach which could support multiple formats/version.
For example:

```csharp
public string ExpositionFormat { get; set; } = "text/plain; charset=utf-8; version=0.0.4";
```

Reply:
It is conceivable that each format may/will have specific requirements that cause the scrape format to be different, even between versions of a specified format. For example, `application/openmetrics-text` requires no extra CR characters, timestamps must be in seconds with a decimal part, and must finish with an `# EOF` line.

Given these restrictions, wouldn't it be better to allow specifying the format using an enum instead, which then lets the serializer decide the content type and all the formatting details?

Alternatively, we can expose the content type property as above, and the serializer can throw (or default back to text/plain) if the exposition format is invalid or not supported by the serializer.

A 3rd option is to expose an abstract class or interface that allows for different formats to be implemented independently. eg.

```c#
public PrometheusSerializer { get; set; } = new TextPlainPrometheusSerializer();
```

Reply:
> It is conceivable that each format may/will have specific requirements that cause the scrape format to be different, even between versions of a specified format. For example, `application/openmetrics-text` requires no extra CR characters, timestamps must be in seconds with a decimal part, and must finish with an `# EOF` line.
> 
> Given these restrictions, wouldn't it be better to allow specifying the format using an enum instead, which then lets the serializer decide the content type and all the formatting details?

I think it depends. The immediate benefit of using enum is the strong type and editor/IDE support (e.g. auto completion). The potential downsides are:
1. enums are part of the public API, making it hard to deprecate anything unless we do a major version bump.
2. if the logic allows more flexibility (e.g. specifying an encoding which is not utf8), enum doesn't fit well.

> Alternatively, we can expose the content type property as above, and the serializer can throw (or default back to text/plain) if the exposition format is invalid or not supported by the serializer.

Makes sense to me if we throw exceptions when the specified exposition format is not supported.

> A 3rd option is to expose an abstract class or interface that allows for different formats to be implemented independently. eg.
> 
> ```cs
> public PrometheusSerializer { get; set; } = new TextPlainPrometheusSerializer();
> ```

I think this makes the configuration story a bit more complex. It also shares the potential issue with enums - making it hard to deprecate anything without a major version bump.

Reply:
Understood, let me have a go at exposing the content type and working that into the serializer.

Reply:
I've had a closer look at this, and I'm not sure allowing the content type to be configurable makes sense. The content type is switchable based on the request `Accept` headers. For example, if you request the `/metrics` endpoint from the browser, it will return `text/plain` which will display correctly, whereas forcing the `application/openmetrics-text` content type will actually try and download the file instead of displaying it (at least using Chrome). So we need to maintain the same level of usability while allowing the content type to switch for Prometheus.

From Prometheus, it will always send `application/openmetrics-text` in `Accept`, but will process either `text/plain` or `application/openmetrics-text` based on the returned `Content-Type`. 

There is some precedent for baking this into the code in this way. As far as I can tell the Java version of OTEL works the same way, and doesn't expose any method for disabling or configuring the `application/openmetrics-text` content type. I think the Go version also works this way, though I'm not a Golang expert.

For now I've removed the flags for enabling/disabling the OpenMetrics format, and just let the middleware/httplistener decide which format to send back based on the `Accept` header.

Hope this makes sense...